src/spi/message/MessageClient.ts

```diff
@@ -3,28 +3,114 @@ import {
     SlackMessage,
 } from "@atomist/slack-messages/SlackMessages";
 import * as _ from "lodash";
-import { ScriptedFlushable } from "../../internal/common/Flushable";
 import { metadataFromInstance } from "../../internal/metadata/metadataReading";
 
 /**
  * Implemented by classes that can send bot messages, whether to
  * channels or individuals, including actions and updates.
  */
-export interface MessageClient extends ScriptedFlushable<MessageClient> {
+export interface MessageClient {
 
-    respond(msg: string | SlackMessage, options?: MessageOptions): Promise<any>;
+    /**
+     * Send a response back to where this command request originated.
+     * @param msg
+     * @param {MessageOptions} options
+     * @returns {Promise<any>}
+     */
+    respond(msg: any,
+            options?: MessageOptions): Promise<any>;
+
+    /**
+     * Send a message to any given destination.
+     * @param msg
+     * @param {Destination | Destination[]} destinations
+     * @param {MessageOptions} options
+     * @returns {Promise<any>}
+     */
+    send(msg: any,
+         destinations: Destination | Destination[],
+         options?: MessageOptions): Promise<any>;
+}
 
-    addressUsers(msg: string | SlackMessage, userNames: string | string[], options?: MessageOptions): Promise<any>;
+/**
+ * MessageClient to send messages to the default Slack team.
+ *
+ * Note: This implementation is deprecated in favor of MessageClient.
+ */
+export interface SlackMessageClient {
 
-    addressChannels(msg: string | SlackMessage, channelNames: string | string[],
+    addressUsers(msg: string | SlackMessage,
+                 users: string | string[],
+                 options?: MessageOptions): Promise<any>;
+
+    addressChannels(msg: string | SlackMessage,
+                    channels: string | string[],
                     options?: MessageOptions): Promise<any>;
+}
+
+/**
+ * Basic message destination.
+ */
+export interface Destination {
+
+    userAgent: string;
+}
+
+/**
+ * Message Destination for Slack.
+ */
+export class SlackDestination implements Destination {
+
+    public userAgent: string = "slack";
```

Comment:
Perhaps this should be `static`? That way it could be used in comparisons like https://github.com/atomist/automation-client-ts/pull/195/files#diff-df59de834cabfa5cb6c0e214e806d03fR46